### ISSUE TITLE
[Feature Request] Option to Enable / Disable searching for a movie when added from a list.

### ISSUE THREAD

**Issue by rkmfl** [Radarr/Radarr#2950]
**Describe the bug**
```
Settings > Lists > Clean Library Level > 
```
**Broken/Bug:** 
Each option (from that select list) that to stop monitoring a movie release.

**To Reproduce**
1. Choose any setting with an "unmonitor" option ("Keep but Unmonitor" or "Remove & Keep Files" or "Remove & Delete Files" etc)
2. Configure a Radarr List (use any integrated option: TMDB, Trakt, etc)
3. Movie-A on list (eg TMDB List) is downloaded
4. Later ... Movie-A is removed from list
5. Later ... Movie-A returns to list
6. Movie is re-downloaded (even when that movie already exists, was previously downloaded by Radarr, and is un-touched)

**Expected behavior**
* If a movie _does exist_ ... _it should not be re-downloaded_
* If a movie does _not_ exist, and the setting indicates it will not be re-downloaded ... _it should not be re-downloaded_.

**Desktop (please complete the following information):**
 - Windows (various, Win8.1 Pro to Win10 Pro)
 - Radarr v0.2.0.1120

**Similar issue that was marked stale (but that issue **_does not_** cover the full breadth of the problem, the issue is not limited to "keep but unmonitor" **
@jusher00 created #2906 and @chuckkay created #2584 

**Note**
_This process will occur as many times as the movie falls-off, then re-appears in the list._  

For example, if it happens 5 times, a movie will be re-downloaded 5 times.  The quality is met, this is not an issue with intentional re-downloads; it happens on all movies that are dropped from a Radarr List, then re-appear again in the future.  

I maintain three different Radarr servers for friends/family, every one of them does this.

**Log Files: 
I've cleaned-up a log file, and included only the relevant entries, to make it easier for you to see exactly what's happening:**
```
# Movie missing from TMDB list -- Removed from Library at 2pm
18-7-31 14:24:49.8|Info|TMDbImport|Importing TMDb movies from: https://api.themoviedb.org/3/discover/movie?api_key=<<private>>&primary_release_date.gte=2018-04-30&primary_release_date.lte=2018-07-31&vote_count.gte=190&vote_average.gte=5.2&with_genres=28|12|35|14|878|53|9648&without_genres=27|10402|37&with_original_language=en&page=1
18-7-31 14:24:52.3|Info|NetImportSearchService|[tt0974015][Justice League (2017)] was in your library, but not found in your lists --> Removing from library (keeping files)

# Movie re-appears in TMDB list -- Downloaded a second time (Movie _already_ exists)
18-7-31 18:54:04.1|Info|PreDBService|Starting PreDB Sync
18-7-31 19:25:10.5|Info|TMDbImport|Importing TMDb movies from: https://api.themoviedb.org/3/discover/movie?api_key=<<private>>&primary_release_date.gte=2018-04-30&primary_release_date.lte=2018-07-31&vote_count.gte=190&vote_average.gte=5.2&with_genres=28|12|35|14|878|53|9648&without_genres=27|10402|37&with_original_language=en&page=1
18-7-31 19:25:13.0|Info|NetImportSearchService|[tt4649466][Kingsman: The Golden Circle (2017)] was in your library, but not found in your lists --> Removing from library (keeping files)
18-7-31 19:25:14.9|Info|MovieService|Adding Movie [tt0974015][Justice League (2017)] Path: [<<PATH>>\Movies\MoviesNew\Justice League (2017)]
18-7-31 19:25:15.1|Info|MediaCoverService|Downloading Poster for [tt0974015][Justice League (2017)] http://image.tmdb.org/t/p/original/eifGNCSDuxJeS1loAXil5bIGgvC.jpg
18-7-31 19:25:15.3|Info|RefreshMovieService|Updating Info for Justice League
18-7-31 19:25:15.9|Info|DiskScanService|Scanning disk for Justice League
18-7-31 19:25:16.0|Info|MediaCoverService|Downloading Fanart for [tt0974015][Justice League (2017)] http://image.tmdb.org/t/p/original/o5T8rZxoWSBMYwjsUFUqTt6uMQB.jpg
18-7-31 19:25:16.4|Info|MovieService|Linking [Justice League (2017).mkv] > [[tt0974015][Justice League (2017)]]
18-7-31 19:25:16.4|Info|DiskScanService|Completed scanning disk for Justice League
18-7-31 19:25:16.4|Info|MovieScannedHandler|[Justice League] was recently added, performing post-add actions
18-7-31 19:25:16.7|Info|ExistingMetadataImporter|Found 6 existing metadata files
18-7-31 19:25:16.9|Info|NzbSearchService|Searching 2 indexers for [Justice League]
18-7-31 19:25:17.0|Info|ExistingSubtitleImporter|Found 0 existing subtitle files
18-7-31 19:25:17.1|Info|ExistingOtherExtraImporter|Found 0 existing other extra files
18-7-31 19:25:17.1|Info|ExistingExtraFileService|Found 0 extra files
18-7-31 19:25:19.9|Info|DownloadDecisionMaker|Processing 204 releases
18-7-31 19:25:23.5|Info|Sabnzbd|Adding report [Justice.League.2017.BDRip.x264-SPARKS] to the queue.
18-7-31 19:25:23.8|Info|DownloadService|Report sent to SAB. Justice.League.2017.BDRip.x264-SPARKS
18-7-31 19:25:23.9|Info|MovieSearchService|Movie search completed. 1 reports downloaded.
18-7-31 19:28:11.6|Info|MovieService|Linking [Justice League (2017).mkv] > [[tt0974015][Justice League (2017)]]
```

**Comment by galli-leo** [comment]
The problem is, that when a movie reappears on the list, Radarr will do a search. An automatic search will ignore the monitored status of a movie.

The solution here, would be to add an option to each list, whether to auto search on add or not.

**Comment by rkmfl** [comment]
@galli-leo - 
You changed the title to something that is actually different from the issue I explained.  I can see it's just a simple misunderstanding of what I explained, I need to clarify one key point:

I think you may not have understood that the movie being searched/downloaded **still exists** ... both on the storage ... and in Radarr's DB.  

In other words, when choosing "**Keep but unmonitor" -- it will exist on disk -AND- Radarr's DB.  When choosing "**Remove and Keep files**" it's removed from Radarr's DB but still on disk).

Simply put -- you definitely do **not** need a _new feature_ as a workaround.  

I think we just need the existing functionality fixed, where Radarr does _not_ download movies that already exist in Radarr's DB.  Or in the case of "Remove and Keep files", Radarr doesn't re-download something it has in the past.

**Comment by galli-leo** [comment]
The option I would add as explained above, would fix both of these issues.

Keep but unmonitor actually works now on the latest nightly.

Remove and keep files would only work, if automatic searching is disabled for the list, since Radarr won't know about the existing file until it has done a search.

**Comment by rkmfl** [comment]
It just seems like it would be easier to remove complexity than to add it.

In other words, leave searches as a **scheduled daemon** rather than triggered by list-scrape.  As long as this is done, everything always "just works" because it's one single approach.  In addition, if you do prefer to prompt a search as a result of new list-scrape-results, just trigger the full step-by-step process (as it's normally set to run, where the movies are imported before searches are done).

Just to be clear, I'm not concerned about any of these, just trying to offer a solution that is easiest for you.  Because for my own purposes, if "Keep but Unmonitor" was fixed, I'm golden!

**Comment by Qstick** [comment]
@galli-leo the proposed option would make this same as #1476 right

**Comment by rkmfl** [comment]
@Qstick - I don't see how this would be the same as #1476 

In the case of 1476, it's proposed to have an **option** whether to search or not.

In this case, movies that are _already downloaded, are being **re-downloaded**_, because they slipped-off an RSS list for a period.

In this case we need to make sure that when movies are found in a list, they are not searched until after Radarr is aware of what already exists.

**Comment by Qstick** [comment]
@galli-leo proposed solution and the new title of this issue is identical to the other issue, that’s what I’m referring to. An option to enable/disable search in lists.

**Comment by rkmfl** [comment]
@Qstick - good point, you're right, I had forgotten about the title change.

I'm still confused how that "option" would fix the error.  If someone switches-off searching, sure that avoids the issue .. but it also stops them from downloading movies they want.

In short, it's not a fix, but a workaround.  The better solution is for the functionality to work properly so that regardless of the setting, movies that already exist are not _re-downloaded_, when they re-appear in "lists".

**Comment by Qstick** [comment]
I think 1504 may fix this (the original issue)
